packages/ra-supabase-core/src/authProvider.ts

```diff
@@ -1,18 +1,28 @@
 import { AuthProvider, UserIdentity } from 'ra-core';
-import { SupabaseClient, User } from '@supabase/supabase-js';
+import { Provider, SupabaseClient, User } from '@supabase/supabase-js';
 
 export const supabaseAuthProvider = (
     client: SupabaseClient,
     { getIdentity }: SupabaseAuthProviderOptions
 ): SupabaseAuthProvider => ({
-    async login({ email, password }: { email: string; password: string }) {
-        const { error } = await client.auth.signInWithPassword({
-            email,
-            password,
-        });
+    async login(params) {
+        const emailPasswordParams = params as LoginWithEmailPasswordParams;
+        if (emailPasswordParams.email && emailPasswordParams.password) {
+            const { error } = await client.auth.signInWithPassword(
+                emailPasswordParams
+            );
 
-        if (error) {
-            throw error;
+            if (error) {
+                throw error;
+            }
+        }
+
+        const oauthParams = params as LoginWithOAuthParams;
+        if (oauthParams.provider) {
+            client.auth.signInWithOAuth(oauthParams);
+            // To avoid react-admin to consider this as an immediate success,
+            // we return a rejected promise that is handled by the default OAuth login buttons
+            return Promise.reject();
         }
         return undefined;
```

Comment:
Shouldn't the login be unsuccessful if no auth mode was matched?

Reply:
I was planning to add magic link support but you're right